runtime/protocol/http-client/common/src/aws/smithy/kotlin/runtime/http/interceptors/OperationTelemetryInterceptor.kt

```diff
@@ -94,6 +96,13 @@ internal class OperationTelemetryInterceptor(
         if (attempts > 1) {
             metrics.rpcRetryCount.add(1L, perRpcAttributes, metrics.provider.contextManager.current())
         }
+
+        val attemptDuration = attemptStart?.elapsedNow() ?: return
+        metrics.rpcAttemptDuration.recordSeconds(attemptDuration, perRpcAttributes, metrics.provider.contextManager.current())
+
+        context.executionContext.getOrNull(EngineAttributes.TimeToFirstByte)?.let { ttfb ->
```

Comment:
You're right we may have an issue if we fail to record TTFB on a retry. We certainly _can_ clear it after reading it, although this makes me wonder if there's a new set of execution context which is scoped to a single attempt and should be cleared when an attempt is ended.

What would we think about a new optional _scope_ for attributes:

```kotlin
interface MutableAttributes : Attributes {
    // ...existing set, remove, and computeIfAbsent methods...

    operator fun <T : Any> set(scope: AttributeScope, key: AttributeKey<T>, value: T)
    fun <T : Any> removeAll(scope: AttributeScope)
}
```

Get operations would find an attribute regardless of its scope but `removeAll` could be used to clear all attributes with a specific scope (e.g., an attempt).

Reply:
Hmm maybe. Seems like it may complicate some things but wouldn't know until I tried. Also raises questions like what attributes are visible? If I set an attribute in one scope but not another are they visible to other scopes? Are they independent (i.e. can they shadow each other)?

Reply:
You're right, this is likely too complicated without a deeper design. I'll just remove it upon reading.